### PR TITLE
#16 - fix safari 10 shadow root scoped style recalc bug

### PR DESCRIPTION
**Pull request by treshugart**

### CODE REVIEW

File: test/unit.js [hunk 1]
```diff
@@ -8,8 +14,55 @@ describe('skatejs-web-components', () => {
...
+      body.appendChild(div);
+      // Wait for the first flow / layout the style element will cause.
+      setTimeout(() => {
+        // Then set the style content.
+        style.textContent = 'p { border: 1px solid rgb(0, 0, 0); }';
+
+        // We must wait until the next reflow / relayout updating the style element causes to then
+        // check if it appolied.
+        setTimeout(() => {
+          expect(window.getComputedStyle(p).border).to.equal('1px solid rgb(0, 0, 0)');
```
Comment (adevnadia) [inline test/unit.js]: Can you maybe check this for the second time? We had quite a few issues when something was working just once when it supposed to work all the time.
Also the empty case  `style.textContent = ''` - that all the styles are removed properly.

Comment (treshugart) [reply test/unit.js]: Sure

Comment (treshugart) [reply test/unit.js]: Done!

Comment (adevnadia) [reply test/unit.js]: Why the callback hell btw? What happened with the `afterMutations`?

Comment (treshugart) [reply test/unit.js]: `afterMutations` is in Skate proper not this repo. This is also a temporary test suite for a temporary fix, so I've opted not to blow the PR out and try and reuse that for the sake of this fix.

**Pull request merged**